src/smartpeak/source/ui/Widget.cpp

```diff
@@ -148,7 +243,560 @@ namespace SmartPeak
     const ImGuiTableFlags table_flags = ImGuiTableFlags_Resizable | ImGuiTableFlags_Hideable | ImGuiTableFlags_Reorderable |
       ImGuiTableFlags_NoBordersInBody | ImGuiTableFlags_Scroll | ImGuiTableFlags_SizingPolicyFixedX | ImGuiTableFlags_NoSavedSettings |
       ImGuiTableFlags_Sortable | ImGuiTableFlags_MultiSortable;
-
+    
+    static ImGuiTableColumnFlags column_0_flags = { ImGuiTableColumnFlags_DefaultSort | ImGuiTableColumnFlags_NoHide };
+    static ImGuiTableColumnFlags column_any_flags = { ImGuiTableColumnFlags_NoHide };
+    
+    static ImGuiTextFilter filter;
+    filter.Draw("Find");
+    
+    // drop-down list for search field(s)
+    static int selected_col = 0;
+    static std::vector<const char*> cols;
+    cols.resize(headers_.size() + 1);
+    for (size_t header_name = 0; header_name < headers_.size() + 1; ++header_name) {
+      if (header_name == 0)
+      {
+        cols[header_name] = "All";
+      }
+      else if (header_name > 0)
+      {
+        cols[header_name] = headers_(header_name - 1).c_str();
+      }
+    }
+    
+    ImGui::Combo("In Column(s)", &selected_col, cols.data(), cols.size() );
+    
+    static ImVector<ImTableEntry> sequence_table_entries;
```

Comment:
Is there anyway that the handling of individual `ImVector<ImTableEntry>` and other filter/sorting objects can be refactored into a set of modular methods?  Is it necessary to instantiate all of them here?

Reply:
It's necessary to instantiate them here as they get manipulated based on user interaction, sorting/searching cannot be fully modularised as they highly depend on `table_id_` and `ImVector<ImTableEntry>` being updated in the draw method

Reply:
Having said that I could encapsulate some of the sub-functionalities in separate methods